src/geometry.rs

```diff
@@ -0,0 +1,188 @@
+//!
+//! Geometry helper functionality.
```

Comment:
I would like to be able to put most of the code here into macros like the other files. For example, Aabb and Aabu is mostly replicated code, and it will also be easier to add wide versions of these types which I definitely want to do.